cmake/DLAF_AddTest.cmake

```diff
@@ -207,10 +207,7 @@ function(DLAF_addTest test_target_name)
     endif()
 
     string(REPLACE ";" "\", \"" PIKA_EXTRA_ARGS_LIST_CAPI "${_PIKA_EXTRA_ARGS_LIST_CAPI}")
-
-    configure_file(
-      ${PROJECT_SOURCE_DIR}/test/include/dlaf_c_test/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h
-    )
+    configure_file(${PROJECT_SOURCE_DIR}/test/include/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h)
```

Comment:
Could you explain what this change is about?

Reply:
This is a leftover for some experimentation with HDF5 input in the, where I moved `config.h.in` from `test/include/dlaf_c_test/` to `tests/include/` to have a single configuration file for all tests.

I did not revert it back because it might make more sense to have `config.h.in` as a general config file, but currently only C API tests use it, so I don't mind reverting the changes if people prefer it.

Reply:
Reverted, since these changes don't belong here and pollute the PR/review process.